bin/prover-client/src/operators/checkpoint.rs

```diff
@@ -59,6 +61,32 @@ impl CheckpointOperator {
             .ok_or(ProvingTaskError::WitnessNotFound)
     }
 
+    /// Retrieves the [`L2BlockId`] for the given `block_num`
+    pub async fn get_l2id(&self, block_num: u64) -> Result<L2BlockId, ProvingTaskError> {
+        let l2_headers = self
+            .cl_client
+            .get_headers_at_idx(block_num)
+            .await
+            .inspect_err(|_| error!(%block_num, "Failed to fetch l2_headers"))
+            .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
+
+        let headers = l2_headers.ok_or_else(|| {
+            error!(%block_num, "No L2 headers found at block height");
+            ProvingTaskError::WitnessNotFound
```

Comment:
I don't think this is the most appropriate error here.

Reply:
Did minor refactoring here.

Ideally we can get rid of entire `fn get_l2id(&self, block_num: u64) -> Result<L2BlockId, ProvingTaskError>` once we have `RpcCheckpointInfo` to have block_ids instead of block_idx on [STR-755](https://alpenlabs.atlassian.net/browse/STR-755)